### ISSUE TITLE
Advanced digitizing tools not working 

### ISSUE THREAD

**Issue by varrieta** [QGEP/QGEP#203]
Has it already been mentionned that the _Advanced Digitizing Toolbar_ isn't woorking when updating the reaches?

That is mostly because of:
- The obj_id that duplicates when you split an entity or copy the attributes, making the changes impossible to commit
- The geometry that becomes a 'Multilinestring' when you continue an entity by reshaping it

**Comment by m-kuhn** [comment]
On a related note, we should probably also ask the user if he wants to insert a new wastewater node at the splitting position (and possibly more like "create new channel"?).

So we could have two approaches:

 * a custom split tool adapted to our needs
 * extend the QGIS split map tool to trigger an event which our plugin can react to and open a dialog etc.

@3nids what do you think about this?

**Comment by varrieta** [comment]
Indeed, the reshape tool seems to work, it wasn't last time.. my bad ! 

Yes it would be useful to be able to directly digitize a new channel.

**Comment by sjib** [comment]
Here are the instruction on how to model a material or diameter change in VSA-DSS
(Wie modelliere ich einen Kaliber- oder Materialwechsel in der VSA-DSS?)

https://www.vsa.ch/interaktiv/faq/datenstruktur-siedlungsentwaesserung/

Dafür steht in der VSA-DSS die Haltung zur Verfügung. Ein Kanal kann ja aus mehreren Haltungen bestehen. Will man einen Kanal (z.B. von Normschacht oben (NS2) nach Normschacht unten (NS1)) infolge Kaliber- oder Materialwechsels unterteilen, wird ein zusätzlicher Abwasserknoten (NS2_NS1) eingefügt. Die erste Haltung geht nun vom Abwasserknoten im Normschacht oben NS2 bis zum Abwasserknoten (NS2_NS1) und hat z.B. das Attribut Haltung.Material = Asbestzement. Die zweite Haltung geht vom Abwasserknoten (NS2_NS1) bis zum Abwasserknoten in Normschacht unten (NS1). Hier kann man nun z.B. das Attribut Haltung.Material = Beton_armiert setzen. Analog können alle anderen Attribute von Haltung verändert werden. 

![csm_materialwechsel_korr_ea38a5ca17](https://cloud.githubusercontent.com/assets/1265865/16869425/00b5214c-4a7d-11e6-8937-d45df4bd4bd9.jpg)

**Comment by urskaufmann** [comment]
With QGIS 3.4 the splitting (and merge) of a reaches with the QGIS-standard-tool works (no more problems with the obj_id). But you have to control manually the fk-connections and the reachpoint levels. One of the two splitted reaches has levels, that are recalculated with the slope of the old reach, the other one has just the old reachpoint-levels (so the slope changes).

**Comment by m-kuhn** [comment]
I think this should be merged with https://github.com/QGEP/QGEP/issues/535
